usaspending_api/awards/models.py

```diff
@@ -225,23 +245,39 @@ def get_or_create_summary_award(piid=None, fain=None, uri=None, awarding_agency=
                 q_kwargs[i[1]] = i[0]
                 if parent_award_id:
                     q_kwargs["parent_award__" + i[1]] = parent_award_id
+                    # parent_award__piid, parent_award__fain, parent_award__uri
                 else:
                     q_kwargs["parent_award"] = None
 
                 # Now search for it
                 # Do we want to log something if the the query below turns up
                 # more than one award record?
+
+                if use_cache:
+                    q_kwargs_fixed = list(q_kwargs.items()) + [('awarding_agency', awarding_agency), ]
+                    q_kwargs_fixed.sort()
+                    summary_award = awards_cache.get(q_kwargs_fixed)
+                    if summary_award:
+                        return summary_award
+
                 summary_award = Award.objects.all().filter(Q(**q_kwargs)).filter(awarding_agency=awarding_agency).first()
                 if summary_award:
+                    if use_cache:
+                        awards_cache.set(q_kwargs_fixed, summary_award)
                     return summary_award
                 else:
                     parent_award = None
                     if parent_award_id:
                         # If we have a parent award id, recursively get/create the award for it
-                        parent_award = Award.get_or_create_summary_award(**{i[1]: parent_award_id, 'awarding_agency': awarding_agency})
+                        parent_award = Award.get_or_create_summary_award(use_cache=use_cache, **{i[1]: parent_award_id, 'awarding_agency': awarding_agency})
                     # Now create the award record for this award transaction
                     summary_award = Award(**{i[1]: i[0], "parent_award": parent_award, "awarding_agency": awarding_agency})
-                    summary_award.save()
+                    if use_cache:
+                        awards_cache.set(q_kwargs_fixed, summary_award)
+                        # when we cache a summary_award whose parent_award is not yet save()ed,
```

Comment:
I just realized that my question-to-self doesn't make sense; since the only way to indicate to `get_or_create_summary_award` that it has a parent is to include the `parent_award_id`, I'm worrying about a situation that can't happen.  So I'm deleting this comment.